### PR TITLE
clean up/rename the md5 stuff

### PR DESCRIPTION
**Pull request by calvinmetcalf**

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
so git didn't catch that I renamed helpers to make-hash

**Comment by calvinmetcalf** [comment]
ok seperated it out into 2 commits so it's easier to compare

**Comment by dcousens** [comment]
LGTM

**Comment by calvinmetcalf** [comment]
woops forgot to lint it

File: make-hash.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  return arr
+}
+
+module.exports = function hash (buf, fn) {
+  var arr = fn(toArray(buf), buf.length * charSize)
+  buf = new Buffer(hashSize)
+  for (var i = 0; i < arr.length; i++) {
+    buf.writeInt32LE(arr[i], i << 2, true)
```
Comment (dcousens) [inline make-hash.js]: `i << 2`, really? Why not `i * 4`

File: make-hash.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+var charSize = 8
+var hashSize = 16
+
+function toArray (buf) {
+  if ((buf.length % intSize) !== 0) {
+    var len = buf.length + (intSize - (buf.length % intSize))
+    buf = Buffer.concat([buf, zeroBuffer], len)
+  }
+
+  var arr = new Array(buf.length >>> 2)
```
Comment (dcousens) [inline make-hash.js]: `buf.length / 4`

Comment (calvinmetcalf) [inline make-hash.js]: Bit operations are faster and automatically round into integers

Comment (dcousens) [inline make-hash.js]: But they aren't readable.
I don't see why the above suggestion couldn't be optimized to do the same.

**Pull request merged**